src/util.js

```diff
@@ -1,5 +1,53 @@
 'use strict';
 
-exports.startsWith = function (haystack, needle) {
+const { Octokit } = require('@octokit/rest');
+
+function startsWith (haystack, needle) {
 	return haystack.substr(0, needle.length) === needle;
 };
+
+async function getDownloadUrl(repo, { version, platform, arch, token }) {
+	const [owner, repository] = repo.split('/');
+	const octokit = new Octokit({ auth: `token ${token}`});
+
+	const { data: release } = await octokit.repos.getReleaseByTag({
+		owner,
+		repository,
+		tag: version,
+	});
+
+	if (!release) {
+		return { error: `Release for ${version} not found` }
+	}
+
+	const { data: assets } = await octokit.repos.listAssetsForRelease({
+		owner,
+		repository,
+		release_id: release.id,
+	});
+
+	const asset = assets.find(asset => {
+		const targetName = `electron-${version}-${platform}-${arch}.zip`;
+		return asset.name === targetName;
+	});
+
+	if (!asset) {
+		return { error: `Release asset for ${version} not found` }
+	}
+
+	const { headers: { location }} = await octokit.repos.getReleaseAsset({
+		owner,
+		repository,
+		asset_id: asset.id,
+		headers: {
+			accept: 'application/octet-stream'

```

Comment:
Seems like https://github.com/octokit/rest.js/issues/967 known with handling private repositories, but since we are not downloading via octokit, can we just do the following

```
const { headers: { browser_download_url }} = await octokit.repos.getReleaseAsset({
		owner,
		repo: repository,
		asset_id: asset.id
});
```

and pass the `browser_download_url` to `@electron/get` ? Is that a supported download mechanism ?

Reply:
it is!